projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-openai/index.js

```diff
@@ -93,6 +93,19 @@ const useSuggestionsFromOpenAI = ( {
 		setIsLoadingCategories( loading );
 	}, [ loading ] );
 
+	useEffect( () => {
+		/*
+		 * Returning a cleanup function that will stop
+		 * the suggestion if it's still rolling.
+		 */
+		return () => {
+			debug( 'Cleaning things up...' );
+			if ( source?.current ) {
+				source?.current?.close();
```

Comment:
I tried to reuse it on a couple different ways:

- by calling it directly; I needed to add it to the `dependencies` array of the `useEffect` hook, and it resulted on the `useEffect` being called every render since the function `stopSuggestion` was redefined on every render, causing the effect to run every time and closing the suggestion early
- by moving it's logic to a function saved by `useCallback`, with the same results

Did I missed something?

Reply:
If I'm not wrong, the hook dependency is altered by the same hook itself. In this case, `loading` dependency is updated by the stopSuggestion() fn.

The fix for it is creating a new hook instance, as we did before, where it doesn't have that dependency:

```es6
useEffect( () => {
  stopSuggestion();
}, [] );
```

Something like ⬆️